ext/spl/spl_directory.c

```diff
@@ -528,9 +528,22 @@ static spl_filesystem_object *spl_filesystem_object_create_type(int num_args, sp
 
 			spl_filesystem_object_get_file_name(source);
 
+			char *open_mode = "r";
+			size_t open_mode_len = 1;
+			zval *resource = NULL;
+
+			if (num_args && zend_parse_parameters(num_args, "|sbr",
+				&open_mode, &open_mode_len, &use_include_path, &resource) == FAILURE
+			) {
+				zend_restore_error_handling(&error_handling);
+				zval_ptr_dtor(return_value);
+				ZVAL_NULL(return_value);
```

Comment:
If you move this code higher (before the ZVAL_OBJ at least), then you can save the zval_ptr_dtor+NULL here.

Reply:
Thanks for the hint!